source/scripts/calendar.js

```diff
@@ -2,8 +2,229 @@
  * A reusable calendar widget
  * @class Calendar
  */
-class Calendar extends HTMLElement {
 
-}
+//&&& need to implement a third "..." redirect button and corresponding modal window
+document.addEventListener("DOMContentLoaded", function () {
+    const monthSelect = document.getElementById("month");
+    const yearInput = document.getElementById("year");
+    const calendarContainer = document.getElementById("calendar");
+    const clearDataButton = document.getElementById('clearBtn');
+    var entryButtons = document.getElementsByClassName('entryButton');
+    
+    // Function to generate calendar
+    function generateCalendar(month, year) {
+      const daysInMonth = new Date(year, month + 1, 0).getDate();
+      const firstDay = new Date(year, month, 1).getDay();
+      
+      //calendarContainer.innerHTML, this generation will need to use local storage to make sure to generate appropriately
+      //make sure to generate with all existing entries
+      let calendarHTML = " <thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody></tbody><tr>";
+  
+      // Add empty cells for days before the first day of the month
+      let count = 0;
+      for (let i = 0; i < firstDay; i++) {
+        calendarHTML += "<td id='emptyCell'></td> ";
+        count++;
+      }
+  
+      // Add cells for each day of the month
+      for (let day = 1; day <= daysInMonth; day++) {
+        if(count % 7 == 0){
+            calendarHTML +=" </tr> <tr>";//creates a new ro
+        }
+        var fill;
+        if(localStorage.getItem(month+'/'+day+'/'+year) == null){
+          fill = day;
+        }
+        else{
+          fill = localStorage.getItem(month+'/'+day+'/'+year);
```

Comment:
Pretty clever usage of localStorage :)

For now we'll keep these but once the backend functions are fully implemented (probably by the end of this week) we'll migrate to that so that we can store everything in files.